src/hedera/hederaEngine.js

```diff
@@ -0,0 +1,347 @@
+
+    let nativeAmount = '0'
+    if (typeof edgeSpendInfo.spendTargets[0].nativeAmount === 'string') {
+      nativeAmount = edgeSpendInfo.spendTargets[0].nativeAmount
+    } else {
+      throw new NoAmountSpecifiedError()
+    }
+
+    const hbar = hedera.Hbar.fromTinybar(nativeAmount)
+    const txnFee = hedera.Hbar.fromTinybar(900000)
+
+    if (!this.client) {
+      throw new Error('this.client was assigned undefined')
+    }
+
+    const transferTx = new hedera.CryptoTransferTransaction(this.client)
+      .addSender(this.client.operator.account, hbar)
+      .addRecipient(publicAddress, hbar)
+      .setTransactionFee(txnFee)
+      .build()
+
+    const txnId = transferTx.getTransactionId()
+    const {
+      account: { shard, realm, account },
+      validStartSeconds,
+      validStartNanos
+    } = txnId
+
+    const txnDate = new Date(
+      validStartSeconds * 1000 + Math.floor(validStartNanos / 1000000)
+    )
+
+    const edgeTransaction: EdgeTransaction = {
+      txid: `${shard}.${realm}.${account}@${validStartSeconds}.${validStartNanos}`, // txid
+      date: txnDate.getTime(), // date
+      currencyCode, // currencyCode
+      blockHeight: 0, // blockHeight
+      nativeAmount: hbar
+        .negated()
+        .asTinybar()
+        .toString(),
+      // UI shows the fee subtracted from the sent amount which doesn't make sense here
+      networkFee: '0', // networkFee
+      ourReceiveAddresses: [], // ourReceiveAddresses
+      signedTx: '', // signedTx
+      otherParams: {
+        fromAddress: this.walletLocalData.publicKey,
+        toAddress: publicAddress,
+        transferTx: transferTx.toBytes()
+      }
+    }
+
+    this.log(
+      `${nativeAmount} ${this.walletLocalData.publicKey} -> ${publicAddress}`
+    )
+    return edgeTransaction
+  }
+
+  async signTx(edgeTransaction: EdgeTransaction): Promise<EdgeTransaction> {
+    if (
+      !edgeTransaction.otherParams ||
+      !edgeTransaction.otherParams.transferTx
+    ) {
+      throw new Error('missing otherParam transferTx')
+    }
+
+    const transferTx = hedera.Transaction.fromBytes(
+      this.client,
+      edgeTransaction.otherParams.transferTx
+    )
+    await transferTx.sign(this.privateKey)
+
+    return {
+      ...edgeTransaction,
+      otherParams: {
+        ...edgeTransaction.otherParams,
+        signedTx: transferTx.toBytes()
+      }
+    }
+  }
+
+  async broadcastTx(
+    edgeTransaction: EdgeTransaction
+  ): Promise<EdgeTransaction> {
+    if (!edgeTransaction.otherParams || !edgeTransaction.otherParams.signedTx) {
+      throw new Error('InvalidTransactionParams')
+    }
+
+    try {
+      const { signedTx } = edgeTransaction.otherParams
+      await hedera.Transaction.fromBytes(
+        this.client,
+        signedTx
+      ).executeForReceipt()
+    } catch (e) {
+      this.log(e)
+      throw e
+    }
+    // must be > 0 to not show "Synchronizing"
+    edgeTransaction.blockHeight = 1
```

Comment:
Since Hedera doesn't have a concept of block height, should I override `getBlockHeight()` to return a value greater than 1 or set it in `walletLocalData`?

Reply:
I overrode `getBlockHeight()` to always return `2`.